### PR TITLE
Use absolute paths for the context managers.

### LINKED ISSUE #45: Do not use relative paths in the layer code!
@jamesbeedy found a bug where he was using another layer that was calling `os.chdir()` methods in the layer-ruby (which is completely valid). https://github.com/battlemidget/juju-layer-ruby/pull/6

This broke the layer-tls because all of the context manager calls are relative paths assuming current working directory is the charm directory. 

We can not make this assumption, and need to fix the relative links when using the `chdir` context manager.

### PR DESCRIPTION
**Pull request by mbruzek**
Fixes #45 

I noticed that most of the `chdir` context managers were using relative paths. Make them use absolute paths.

### CODE REVIEW

**Comment by mbruzek** [comment]
@jamesbeedy can you have a look at this PR to see if the changes are legit?

File: reactive/tls.py [hunk 1]
```diff
@@ -24,12 +24,16 @@
 @when_not('easyrsa installed')
 def install():
     '''Install the easy-rsa software that is required for this layer.'''
-    if os.path.isdir('easy-rsa'):
-        shutil.rmtree('easy-rsa')
+    # Create an absolute path to easy-rsa that is not affected by cwd.
+    easy_rsa_directory = os.path.join(hookenv.charm_dir(), 'easy-rsa')
```
Comment (chuckbutler) [inline reactive/tls.py]: Nice, i like that you've explicitly called the path here

**Comment by chuckbutler** [comment]
I like that you've refactored paths to be abs paths. Are we certain that these paths won't be changing in the near term?  Minor comment, as we would have had the exact same problem with the relative bathing we were using.

+1 LGTM

**Comment by jamesbeedy** [comment]
@mbruzek niceee +1

**Pull request merged**